pkg/mattermost/mattermost.go

```diff
@@ -152,32 +151,61 @@ func GenerateDeployment(mattermost *mattermostv1alpha1.ClusterInstallation, dbIn
 				},
 				Command: []string{
 					"sh", "-c",
-					"until curl --max-time 5 $DB_CONNECTION_CHECK_URL; do echo waiting for mysql; sleep 5; done;",
+					"until curl --max-time 5 $DB_CONNECTION_CHECK_URL; do echo waiting for database; sleep 5; done;",
 				},
 			})
 		}
 	} else {
+		mysqlName := utils.HashWithPrefix("db", mattermost.Name)
+
 		masterDBEnvVar.Value = fmt.Sprintf(
-			"mysql://%s:%s@tcp(%s-mysql-master.%s:3306)/%s?charset=utf8mb4,utf8&readTimeout=30s&writeTimeout=30s",
-			dbInfo.UserName, dbInfo.UserPassword, mysqlName, mattermost.Namespace, dbInfo.DatabaseName,
+			"mysql://$(MYSQL_USERNAME):$(MYSQL_PASSWORD)@tcp(%s-mysql-master.%s:3306)/%s?charset=utf8mb4,utf8&readTimeout=30s&writeTimeout=30s",
+			mysqlName, mattermost.Namespace, dbInfo.DatabaseName,
 		)
 
-		envVarDB = append(envVarDB, corev1.EnvVar{
-			Name: "MM_SQLSETTINGS_DATASOURCEREPLICAS",
-			Value: fmt.Sprintf(
-				"%s:%s@tcp(%s-mysql.%s:3306)/%s?readTimeout=30s&writeTimeout=30s",
-				dbInfo.UserName, dbInfo.UserPassword, mysqlName, mattermost.Namespace, dbInfo.DatabaseName,
-			),
-		})
+		operatorEnv := []corev1.EnvVar{
+			{
+				Name: "MYSQL_USERNAME",
+				ValueFrom: &corev1.EnvVarSource{
+					SecretKeyRef: &corev1.SecretKeySelector{
+						LocalObjectReference: corev1.LocalObjectReference{
+							Name: dbInfo.SecretName,
+						},
+						Key: "USER",
+					},
+				},
+			},
+			{
+				Name: "MYSQL_PASSWORD",
+				ValueFrom: &corev1.EnvVarSource{
+					SecretKeyRef: &corev1.SecretKeySelector{
+						LocalObjectReference: corev1.LocalObjectReference{
+							Name: dbInfo.SecretName,
+						},
+						Key: "PASSWORD",
+					},
+				},
+			},
+			{
+				Name: "MM_SQLSETTINGS_DATASOURCEREPLICAS",
+				Value: fmt.Sprintf(
+					"$(MYSQL_USERNAME):$(MYSQL_PASSWORD)@tcp(%s-mysql.%s:3306)/%s?readTimeout=30s&writeTimeout=30s",
```

Comment:
maybe we can add this in the secret and then avoid to make the data public available in the deployment. what do you think?

Reply:
i will explain to you in a call, but lgtm as is